### PR TITLE
Function merge_padding_and_attention_mask does not return an output with the desired shape when both padding and attention masks are given

### PR DESCRIPTION
**Pull request by abodinier**
Function merge_padding_and_attention_mask does not return an output with the desired shape when both padding and attention masks are given.

See the linked issue here: https://github.com/keras-team/keras-nlp/issues/783#issue-1600168236

Currently, the TransformerEncoder layers breaks if we call it with both padding and attention masks.

Changes: 
- Remove duplicated `tf.newaxis`
- Create a check function to control the shapes of the padding and attention masks
- Update unit tests to make sure that the error is raised if we give `merge_padding_and_attention_mask` masks with bad shapes.

### CODE REVIEW

**Comment by atharvapurdue** [comment]
Hi @abodinier , great job!
I think there is small change required. `merged_mask` is assigned but never used. The check should pass after that

File: keras_nlp/layers/transformer_layer_utils.py [hunk 1]
```diff
@@ -18,6 +18,22 @@
...
 
 
+def _check_masks_shapes(inputs, padding_mask, attention_mask):
+    mask = padding_mask
+    if hasattr(inputs, "_keras_mask") and mask is None:
+        mask = inputs._keras_mask
+    if mask is not None:
+        if mask._rank() != 2:
+            raise ValueError(
+                f"padding_mask should be of size [B, T], found {mask.shape}"
```
Comment (mattdangerw) [inline keras_nlp/layers/transformer_layer_utils.py:28]:
```
f"`padding_mask` should be of shape `(batch_size, target_length)`. "
"Received shape `{mask.shape}`."
```

File: keras_nlp/layers/transformer_layer_utils.py [hunk 2]
```diff
@@ -18,6 +18,22 @@
...
+            )
+    if attention_mask is not None:
+        if attention_mask._rank() != 3:
+            raise ValueError(
+                f"attention_mask should be of size [B, T, T], found {attention_mask.shape}"
```
Comment (mattdangerw) [inline keras_nlp/layers/transformer_layer_utils.py:33]:
```
f"`attention_mask` should be of shape "
"`(batch_size, target_length, source_length)`. "
"Received shape `{mask.shape}`."
```

**Review by mattdangerw — APPROVED** [review]
This looks great! Some minor comments, this also still has a few minor formatting errors (`./shell/format.sh`), and needs to be merged with master.

Thanks!
